client/src/components/Form/Elements/FormDialog.vue

```diff
@@ -0,0 +1,68 @@
+<script setup lang="ts">
+import { FontAwesomeIcon } from "@fortawesome/vue-fontawesome";
+import { library } from "@fortawesome/fontawesome-svg-core";
+import { faFolderOpen } from "@fortawesome/free-solid-svg-icons";
+import DataDialog from "../../DataDialog/DataDialog.vue";
+import { onMounted } from "vue";
+import { getCurrentGalaxyHistory } from "../../../utils/data";
+import { reactive } from "vue";
+
+library.add(faFolderOpen);
+
+interface DataDialogProps {
+    id: string;
+    multiple: boolean;
+    value?: Array<string> | string;
+}
+
+const props = withDefaults(defineProps<DataDialogProps>(), {
+    multiple: false,
+});
+
+const title = "Browse Datasets";
+const historyID: string = "";
+const localHistoryId = reactive({
+    value: historyID,
+});
+
+onMounted(() => {
+    getCurrentGalaxyHistory().then((historyId) => {
+        localHistoryId.value = historyId;
+    });
+});
+
+const initialODD = false;
+const openDataDialog = reactive({
```

Comment:
Why does this use `reactive` instead of `ref`?

Reply:
When I try with a ref instead I get this ts error:
```TS2322: Type 'boolean' is not assignable to type 'Ref<boolean>'.```

Reply:
I tried to reproduce this error, but could not. Try changing the line in question to:

```ts
const openDataDialog = ref(false);
```

If this does not fix it, please push your changes so we can have a closer look.

Also note that, counterintuitively, ref needs to be used in two different manners, depending on if you are using it in javascript or vue templates. In the templates, refs get pre-processed and the `.value` is omitted. Refs are essentially treated like plain values here. In code however, you do need to use `.value`, to access the underlying value of a ref.

Reply:
^ ah, that makes sense to me and does seem to clear up the error that I came across. Thank you!